apps/nextjs-website/src/lib/api.ts

```diff
@@ -34,10 +37,37 @@ export async function getApi(productSlug?: string) {
 export async function getGuide(
   productSlug?: string,
   productGuidePage?: ReadonlyArray<string>
-) {
+): Promise<GuidePage> {
   const guidePath = productGuidePage?.join('/');
   const path = `/${productSlug}/guides/${guidePath}`;
-  const props = manageUndefined(guides.find(({ page }) => page.path === path));
+
+  const guide =
+    guides.find(({ page }) => page.path === path) ?? getMainGuide(path);
+  const props = manageUndefined(guide);
+
+  const isPathWithMainVersion =
+    props.version.main && !path.startsWith(props.version.path);
+
+  // We need to redirect to the main version if the user is trying to access to a url without version
+  // Example: /sanp/specifiche-attuative-del-nodo-dei-pagamenti-spc/changelog
+  // We need to redirect to /sanp/3.6.0/specifiche-attuative-del-nodo-dei-pagamenti-spc/changelog
+
+  if (isPathWithMainVersion) {
+    // Check if the current path has a version in it
+    const hasVersionInPath = (guidePath ?? '').match(VERSION_REGEX);
+    const sliceFrom = hasVersionInPath ? 2 : 1;
+    const to = [
+      props.version.path,
+      ...(productGuidePage ?? []).slice(sliceFrom),
+    ].join('/');
+
+    // Check if the guide exists
+    const guide = await getGuide(productSlug, getProductGuidePath(to));
+    return {
+      ...guide,
+      redirect: true,
+    };
+  }
```

Comment:
I think this is the correct way to manage redirect to main version. Unfortunately is not possible now without standalone deploy of next. 
Only temporary i would move this logic into a useEffect inside a dedicated "GuidPageWrapper" component that wrap guide page (we need a wrapper beacuse pages cannot use "useEffect" hook itself).

Reply:
One other solution would be to create a duplicate of the latest version of the guide and provide it with both the versioned and non-versioned paths. For example, if the latest version of the guide is "io-guida-tecnica/v3.0", then it should be generated with both `io-guida-tecnica/v3.0` and `io-guida-tecnica` paths. It would all happen at build-time